registry/dynamodb.go

```diff
@@ -59,21 +65,38 @@ type DynamoDBRegistry struct {
 	cacheInterval           time.Duration
 }
 
+const dynamodbAttributeMigrate = "dynamodb/needs-migration"
+
 // NewDynamoDBRegistry returns a new DynamoDBRegistry object.
-func NewDynamoDBRegistry(provider provider.Provider, ownerID string, dynamodbAPI DynamoDBAPI, table string, cacheInterval time.Duration) (*DynamoDBRegistry, error) {
+func NewDynamoDBRegistry(provider provider.Provider, ownerID string, dynamodbAPI DynamoDBAPI, table string, txtPrefix, txtSuffix, txtWildcardReplacement string, managedRecordTypes []string, txtEncryptAESKey []byte, cacheInterval time.Duration) (*DynamoDBRegistry, error) {
 	if ownerID == "" {
 		return nil, errors.New("owner id cannot be empty")
 	}
 	if table == "" {
 		return nil, errors.New("table cannot be empty")
 	}
 
+	if len(txtEncryptAESKey) == 0 {
+		txtEncryptAESKey = nil
+	} else if len(txtEncryptAESKey) != 32 {
+		return nil, errors.New("the AES Encryption key must have a length of 32 bytes")
```

Comment:
I think we should have exported error vars to test them in the top level code.

Reply:
Why would the top-level code want to test them? These are all "configuration is bad; refuse to start" errors.

Incidentally, this validation code is copied from the TXT registry.